LoopKit/StatusStore.swift

```diff
@@ -0,0 +1,188 @@
+//
+//  StatusStore.swift
+//  LoopKit
+//
+//  Created by Darin Krauss on 10/14/19.
+//  Copyright © 2019 LoopKit Authors. All rights reserved.
+//
+
+import Foundation
+import HealthKit
+
+public protocol StatusStoreDelegate: AnyObject {
+    
+    /**
+     Informs the delegate that the status store has updated status data.
+     
+     - Parameter statusStore: The status store that has updated status data.
+     */
+    func statusStoreHasUpdatedStatusData(_ statusStore: StatusStore)
+    
+}
+
+public class StatusStore {
```

Comment:
nit: "Status" is pretty vague. Would `LoopStatus` or `DosingDecisionData` be better?

Reply:
Which would you prefer?

Reply:
I think `DosingDecision` as the data object, and `DosingDecisionStore` as the store name could help scope the content better to help prevent this from becoming more of a dumping ground.

Reply:
I'm open to other suggestions.  I just see "status" as something that, if not scoped, would be interpreted to mean *any* status of the entire Loop app, which does change more often than just dosing decision points.

Reply:
Done.